apm-sniffer/apm-sdk-plugin/elasticsearch-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v6/interceptor/IndicesClientCreateMethodsInterceptor.java

```diff
@@ -0,0 +1,79 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.apache.skywalking.apm.plugin.elasticsearch.v6.interceptor;
+
+import static org.apache.skywalking.apm.agent.core.conf.Config.Plugin.Elasticsearch.TRACE_DSL;
+
+import java.lang.reflect.Method;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+import org.apache.skywalking.apm.plugin.elasticsearch.v6.ESTypeEnum;
+import org.apache.skywalking.apm.plugin.elasticsearch.v6.RestClientEnhanceInfo;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+
+/**
+ * @author aderm
+ */
+public class IndicesClientCreateMethodsInterceptor implements InstanceMethodsAroundInterceptor {
+
+    private static final ILog logger = LogManager.getLogger(IndicesClientCreateMethodsInterceptor.class);
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+        Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        CreateIndexRequest createIndexRequest = (CreateIndexRequest)(allArguments[0]);
+
+        RestClientEnhanceInfo restClientEnhanceInfo = (RestClientEnhanceInfo) (objInst
+            .getSkyWalkingDynamicField());
+        if (restClientEnhanceInfo != null) {
+            AbstractSpan span = ContextManager
+                .createExitSpan(Constants.CREATE_OPERATOR_NAME,
+                    restClientEnhanceInfo.getPeers());
+            span.setComponent(ComponentsDefine.REST_HIGH_LEVEL_CLIENT);
+
+            Tags.DB_TYPE.set(span, ESTypeEnum.CREATE.getType());
+            Tags.DB_INSTANCE.set(span, createIndexRequest.index());
+            if (TRACE_DSL) {
+                Tags.DB_STATEMENT
+                    .set(span, createIndexRequest.mappings().utf8ToString());
+            }
+            SpanLayer.asDB(span);
+        }
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+        Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        ContextManager.stopSpan();
+        return ret;
+    }
```

Comment:
I notice some of the interceptors have this condition(`IndicesClientCreateMethodsInterceptor`, `IndicesClientDeleteMethodsInterceptor`), and the others don't (`RestHighLevelClientGetMethodsInterceptor`, `RestHighLevelClientIndexMethodsInterceptor`), are there differencies? if this condition is a must, then you may need it in `afterMethod`

Reply:
Yes, they are not the same. The front is mainly related to index operations. 

>  if this condition is a must, then you may need it in afterMethod
Not understanding the meaning

Reply:
> Yes, they are not the same. The front is mainly related to index operations.
> 
> > if this condition is a must, then you may need it in afterMethod
> 
> Not understanding the meaning

I was asking why some of them have condition `if (restClientEnhanceInfo != null) {` and other don't, if the condition is really needed, you'll need the same condition in `afterMethod`

Reply:
ACK, in aftermethod, `if (restClientEnhanceInfo != null)`  will be added.